ikasaneip/platform/ikasan-h2-backup/src/main/java/org/ikasan/backup/h2/service/H2BackupServiceImpl.java

```diff
@@ -0,0 +1,328 @@
+package org.ikasan.backup.h2.service;
+
+import org.h2.tools.Server;
+import org.ikasan.backup.h2.exception.H2DatabaseValidationException;
+import org.ikasan.backup.h2.exception.InvalidH2ConnectionUrlException;
+import org.ikasan.backup.h2.model.H2DatabaseBackup;
+import org.ikasan.backup.h2.util.H2BackupUtils;
+import org.ikasan.backup.h2.util.H2ConnectionUrlUtils;
+import org.ikasan.spec.flow.Flow;
+import org.ikasan.spec.housekeeping.HousekeepService;
+import org.ikasan.spec.module.Module;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.test.util.TestSocketUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.*;
+import java.text.SimpleDateFormat;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class H2BackupServiceImpl implements HousekeepService {
+
+    private static Logger logger = LoggerFactory.getLogger(H2BackupServiceImpl.class);
+
+    private static final String CREATE_TABLE_SQL = "CREATE TABLE IF NOT EXISTS %s ( \n" +
+        " IDENTIFIER  VARCHAR NOT NULL, \n" +
+        "   CONSTRAINT %s \n" +
+        "     PRIMARY KEY (IDENTIFIER))";
+
+    public static final String BACKUP_DIRECTORY = "db-backup";
+    public static final String TEST_DIRECTORY = BACKUP_DIRECTORY
```

Comment:
Is TEST_Directory used for testing or it will  be used in runtime

Reply:
Test directory is used at runtime to unpack the backup and run the test on the database to prove that the DB is OK. It is then discarded.